lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -425,6 +461,72 @@ _computeLoadAndUnloadItems(const UsdPrim& prim)
 namespace MAYAUSD_NS_DEF {
 namespace ufe {
 
+std::vector<SchemaTypeGroup> UsdContextOps::fSchemaTypeGroups = {};
+
+bool _schemaGroupSortCompare(const SchemaTypeGroup& a, const SchemaTypeGroup& b)
+{
+    return a.name < b.name;
+}
+
+//! \brief Get groups of concrete schema prim types to list dynamically in the UI
+static const std::vector<SchemaTypeGroup> getConcretePrimTypes(bool sorted)
+{
+    std::vector<SchemaTypeGroup> groups;
+
+    // Query all the available types
+    PlugRegistry&    plugReg = PlugRegistry::GetInstance();
+    std::set<TfType> schemaTypes;
+    plugReg.GetAllDerivedTypes<UsdSchemaBase>(&schemaTypes);
+
+    for (auto t : schemaTypes) {
+        if (!UsdSchemaRegistry::IsConcrete(t)) {
+            continue;
+        }
+
+        auto plugin = plugReg.GetPluginForType(t);
+        if (!plugin) {
+            continue;
+        }
+
+        // For every plugin we check if there's a nice name registered and use that instead
+        auto plugin_name = plugin->GetName();
+        auto name_itr
+            = std::find(kSchemaPluginNames.begin(), kSchemaPluginNames.end(), plugin_name);
+        if (name_itr != kSchemaPluginNames.end()) {
+            plugin_name = kSchemaNiceNames[name_itr - kSchemaPluginNames.begin()];
+        }
+
+        // We don't list empty names. This allows hiding certain plugins too.
+        if (plugin_name.empty()) {
+            continue;
+        }
+
+        auto type_name = UsdSchemaRegistry::GetConcreteSchemaTypeName(t);
```

Comment:
Unfortunately I believe that this method was only added to USD in v20.08. We support back to USD v20.02 so is there a way you can workaround this in older versions? If not, in the worst case you can #ifdef all the new code and only compile it  if PXR_VERSION >= 2008